src/TMxpMusicTagHandler.cpp

```diff
@@ -0,0 +1,90 @@
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "TMedia.h"
+#include "TMxpMusicTagHandler.h"
+#include "TMxpClient.h"
+
+TMxpTagHandlerResult TMxpMusicTagHandler::handleStartTag(TMxpContext& ctx, TMxpClient& client, MxpStartTag* tag)
+{
+    TMediaData mediaData {};
+
+    mediaData.setMediaFileName(tag->getAttributeValue("FName"));
+    mediaData.setMediaType(TMediaData::MediaTypeMusic);
+
+    QString volume = tag->getAttributeValue("V");
+    if (!volume.isEmpty()) {
+        mediaData.setMediaVolume(volume.toInt());
+
+        if (mediaData.getMediaVolume() == TMediaData::MediaVolumePreload) {
+            // Support preloading
+        } else if (mediaData.getMediaVolume() > TMediaData::MediaVolumeMax) {
+            mediaData.setMediaVolume(TMediaData::MediaVolumeMax);
+        } else if (mediaData.getMediaVolume() < TMediaData::MediaVolumeMin) {
+            mediaData.setMediaVolume(TMediaData::MediaVolumeMin);
+        }
+    }
+
+    QString loops = tag->getAttributeValue("L");
+    if (!loops.isEmpty()) {
+        mediaData.setMediaLoops(loops.toInt());
+
+        if (mediaData.getMediaLoops() < TMediaData::MediaLoopsRepeat || mediaData.getMediaLoops() == 0) {
+            mediaData.setMediaLoops(TMediaData::MediaLoopsDefault);
+        }
+    }
+
+    QString priority = tag->getAttributeValue("P");
+    if (!priority.isEmpty()) {
+        mediaData.setMediaPriority(priority.toInt());
+
+        if (mediaData.getMediaPriority() > TMediaData::MediaPriorityMax) {
+            mediaData.setMediaPriority(TMediaData::MediaPriorityMax);
+        } else if (mediaData.getMediaPriority() < TMediaData::MediaPriorityMin) {
+            mediaData.setMediaPriority(TMediaData::MediaPriorityMin);
+        }
+    }
+
+    QString musicContinue = tag->getAttributeValue("C");
+    if (!musicContinue.isEmpty()) {
+        if (musicContinue.toInt() == 0) {
+            mediaData.setMediaContinue(TMediaData::MediaContinueRestart);
+        } else {
+            mediaData.setMediaContinue(TMediaData::MediaContinueDefault);
+        }
+    }
+
+    QString type = tag->getAttributeValue("T");
+    if (!type.isEmpty()) {
+        mediaData.setMediaTag(type.toLower());
+    }
+
+    QString url = tag->getAttributeValue("U");
+    if (!url.isEmpty()) {
+        mediaData.setMediaUrl(url);
+    }
+
```

Comment:
Don't you need to set the setMediaProtocol before calling play?

I had to add `mediaData.setMediaProtocol(TMediaData::MediaProtocolMSP);` somewhere here to make it work.

Reply:
I do, thanks!